### DISCUSSION TITLE
Valkey support

### DISCUSSION THREAD

**Discussion by mariojsnunes** [OrchardCMS/OrchardCore#15913] [Q&A] [answered]
With the recent events around Redis, I wonder if the current OrchardCore.Redis module is compatible with Valkey, and if not, what would be the effor to make it compatible/create OrchardCore.Valkey

**Comment by Piedone** [comment, marked as answer]
Since Valkey is compatible with Redis, OC shouldn't have any issues.

BTW for those wondering, Redis is now dual-licensed: https://redis.io/blog/redis-adopts-dual-source-available-licensing/.

**Comment by avifenesh** [comment]
Hi, what would be the case if compatibility is getting apart?

**Reply by Piedone** [reply]
Then we'd need to copy the Redis module, and create a Valkey one. Are you aware of any differences in the API between the two projects? The internal data format is different, so migrating between the two if you want to keep your data is non-trivial, but since we only use Redis as a cache in OC, that shouldn't really concern us.

**Reply by avifenesh** [reply]
ATM, at least from Valkey side there is strong effort to keep being compatible. But this is possible until Redis 7.2, from then on, Valkey folks won't be going to check the codebase for legal reason.
The data structure is being changed, but the protocol is still the same, so this is not an issue currently, but features are starts to being divergent. 
Some can be significant to users.
For example, if you are a user of some cloud vendor, Valkey AZ affinity is worth a lot of money for you. Data transfer (in AWS at least) is free in the same AZ. And on the way it's also increase performance (probably). 
But to my understanding, those kinds of Valkey specific API are not available to use in configuration using OC.

**Reply by Piedone** [reply]
Yep. So for now, for OC caching there shouldn't be any difference.
